File: corpus/cs1/0008-esquerda-volver/solutions/a-optimal/solve_a.py
import sys

def solve():
    # Mapeamento: 0: N, 1: L, 2: S, 3: O
    dirs = ['N', 'L', 'S', 'O']
    
    lines = sys.stdin.readlines()
    i = 0
    while i < len(lines):
        try:
            line = lines[i].strip()
            if not line:
                i += 1
                continue
                
            N = int(line)
            if N == 0:
                break
            
            i += 1
            if i >= len(lines):
                break
                
            commands = lines[i].strip()
            i += 1
            
            current_dir = 0 # Norte
            
            for cmd in commands:
                if cmd == 'D':
                    current_dir = (current_dir + 1) % 4
                elif cmd == 'E':
                    current_dir = (current_dir - 1 + 4) % 4
            
            print(dirs[current_dir])
            
        except ValueError:
            i += 1
            continue

if __name__ == "__main__":
    solve()

File: corpus/cs1/0008-esquerda-volver/solutions/a-optimal/solve_a.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

// Direções: N, L, S, O
// Mapeamento: 0, 1, 2, 3
// Direita (+1), Esquerda (-1 ou +3)

char get_direction_char(int dir) {
    if (dir == 0) return 'N';
    if (dir == 1) return 'L';
    if (dir == 2) return 'S';
    if (dir == 3) return 'O';
    return '?';
}

void solve() {
    int N;
    while (cin >> N && N != 0) {
        string commands;
        cin >> commands;
        
        int current_dir = 0; // Começa no Norte (0)
        
        for (char cmd : commands) {
            if (cmd == 'D') {
                current_dir = (current_dir + 1) % 4;
            } else if (cmd == 'E') {
                current_dir = (current_dir + 3) % 4;
            }
        }
        
        cout << get_direction_char(current_dir) << endl;
    }
}

int main() {
    solve();
    return 0;
}
